app/src/main/java/com/zulip/android/models/Message.java

```diff
@@ -484,7 +487,7 @@ public String getDisplayRecipient(ZulipApp app) {
             ArrayList<String> names = new ArrayList<>();
 
             for (Person person : people) {
-                if (person.id != app.getYou().id || people.length == 1) {
+                if (!person.getEmail().equals(app.getYou().getEmail()) || people.length == 1) {
```

Comment:
No, the person ID is auto generated so it shouldn't be the same and if you check the dumped databases i posted the person have unique ID's